include/exadg/structure/spatial_discretization/operator.cpp

```diff
@@ -241,6 +241,8 @@ Operator<dim, Number>::compute_scaling_factor_mass(double const scaling_factor_a
   double scaling_factor_mass = scaling_factor_acceleration;
   if(param.weak_damping_active)
   {
+    AssertThrow(param.problem_type == ProblemType::Unsteady,
+                dealii::ExcMessage("Weak damping only well-defined for instationary problems."));
```

Comment:
I think one should assert this when checking the parameters in the respective member function of `Parameters`. The spatial operator then applies the weak damping whenever `param.weak_damping_active == true`. Put differently, `Parameters::check()` aims to avoid (potentially redundant) parameter checks in spatial operators / time integration routines, ...

Reply:
that is absolutely true, sorry I forgot about the discussion we had some weeks ago! will update the PR now.